nifi-api/src/main/java/org/apache/nifi/processor/AbstractProcessor.java

```diff
@@ -21,7 +21,7 @@
 public abstract class AbstractProcessor extends AbstractSessionFactoryProcessor {
 
     @Override
-    public final void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) throws ProcessException {
+    public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) throws ProcessException {
```

Comment:
@ChrisSamo632 Thank you for the review changes and apologies for my late reply!

Regarding this particular question of accessing `ProcessSessionFactory` in a processor that comes from `AbstactProcessor` (indirectly):
Basically, we have the following options (as you already mentioned):

1. Remove the processor from the AWS abstract processors hierarchy and extend `AbstractProcessSessionFactory` processor directly (actually, I would not be against it because the code of the AWS abstract processors is quite complicated/deprecated and the new `ConsumeKinesisStream` does not really need the AWS client object provided by the abstract classes)
2. Inherit `AbstractAWSProcessor` from `AbstractSessionFactoryProcessor`, copy this `onTrigger(ProcessContext, ProcessSessionFactory)` implementation (without the `final` modifier) to the AWS base class and override the method where custom handling of `ProcessSessionFactory` is needed (it would be code duplication and I agree with you that it is not really nice)
3. Remove `final` modifier in `AbstractProcessor` (it looks OK to me but I'm a bit uncertain about it because it would affect the base class of almost all processors and I believe it was a design decision to make the `AbstractProcessor`'s API as simple as possible)

@markap14 What is your opinion on option 3? Can the `final` be removed or it should rather be untouched?

Reply:
RE option 1 - I considered this but decided against it because the new processor does use *some* of the existing abstract AWS processor logic (e.g. Proxy config & Auth). I didn't much fancy re-implementing that separately and also it could be difficult to keep the refactored processor closely aligned enough to future AWS Processor bundle updates in future (people easily not realising that they needed to make the same changes more than once within the bundle).

I think there should be a separate Jira ticket to refactor the processors (or this could be considered as part of the migration to AWS SDK 2.x, a ticket for which I've previously raised) - the AWS bundle could certainly stand some rework/simplkfication, but this PR is already big enough IMO

I've implemented option 3 currently in this PR, reworking to option 2 shouldn't be a big change if decided that's to preferred way forward.